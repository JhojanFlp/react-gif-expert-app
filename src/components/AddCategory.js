import React, { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ addCategory }) => {
  const [value, setValue] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    if (value.trim().length >= 5) {
      addCategory(value);
      setValue('');
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text"
        value={value}
        onChange={(e) => setValue(e.target.value)}
        placeholder="Kimetsu No Yaiba"
      />
    </form>
  )
}

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired
}

export default AddCategory
