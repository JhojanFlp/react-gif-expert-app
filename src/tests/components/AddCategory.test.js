import React from 'react';
import { shallow } from 'enzyme';

import AddCategory from '../../components/AddCategory.js';

describe('Test in AddCategory', () => {
  const addCategory = jest.fn();
  const value = 'Dragon Ball Super';
  let wrapper = shallow(<AddCategory addCategory={addCategory} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory addCategory={addCategory} />);
  })
  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  })
  // test('should change input', () => {
  //   const input = wrapper.find('input');
  //   input.simulate('change', { target: { value } })
  // })
  test('should no add category', () => {
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault(){} });
    expect(addCategory).not.toHaveBeenCalled();
  })
  test('should add category and clean input', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { value }});
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    // expect(addCategory).toHaveBeenCalled();
    expect(addCategory).toHaveBeenCalledTimes(1);
    expect(input.prop('value')).toBe('')
  })
  
})
