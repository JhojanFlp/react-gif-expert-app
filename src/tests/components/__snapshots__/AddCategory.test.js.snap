// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory should render component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Kimetsu No Yaiba"
    type="text"
    value=""
  />
</form>
`;
